documentation/public/user-guide/overview.md

```diff
@@ -0,0 +1,11 @@
+## Design overview
+
+At its core, Slang is a collection of APIs that are meant to analyze the source code, starting with the source code itself and ending with a rich structure that can be reasoned about. This is a departure from the classic approach of "black-box" compilers, which are handed the input and only their output can be observed.
+
```

Comment:
I like `concepts.md`! Renamed in https://github.com/NomicFoundation/slang/pull/674/commits/df0a767f3a16e2cde8cc33e6968f0ab51246902f.

Also separated the Concrete Syntax Tree as the overarching section; if we add more info we can split this further; I wanted to prevent sections with just a single sentence for now.